test/clojure-mode-syntax-test.el

```diff
@@ -60,6 +60,36 @@
       (expect (non-func "^hint " form) :to-be nil)
       (expect (non-func "#macro " form) :to-be nil))))
 
+(describe "clojure-match-next-def"
+  (it "handles vars with metadata"
+    (with-clojure-buffer "
+(def ^Integer a 1)
```

Comment:
Some other test cases that should be analog to this one:

* `(def ^:a a 1)`
* `(def ^::a a 1)`
* `(def ^::a/b a 1)`

Reply:
Done. I've also added `(def ^{:macro true} a 1)`